src/vdc_collection.rs

```diff
@@ -0,0 +1,134 @@
+use super::storage_manager::*;
+use serde_derive::{Deserialize, Serialize};
+use tracing::{info, info_span};
+use uuid::Uuid;
+
+/// Supported credential formats.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialFormat {
+    MsoMdoc,
+    JwtVcJson,
+    LdpVc,
+    Other(String), // For ease of expansion.
+}
+
+/// Supported credential types.
+#[derive(PartialEq, Debug, Serialize, Deserialize)]
+pub enum CredentialType {
+    Iso18013_5_1mDl,
+    VehicleTitle,
+    Other(String), // For ease of expansion.
+}
+
+/// An individual credential.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Credential {
+    id: Uuid,
+    format: CredentialFormat,
+    ctype: CredentialType,
+    payload: Vec<u8>, // The actual credential.
+}
+
+impl Credential {
+    /// Create a new credential.
+    fn new(
+        id: Uuid,
+        format: CredentialFormat,
+        ctype: CredentialType,
+        payload: Vec<u8>,
+    ) -> Credential {
+        Credential {
+            id,
+            format,
+            ctype,
+            payload,
+        }
+    }
+}
+
+/// Verifiable Digital Credential Collection
+///
+/// This is the main interface to credentials.
+pub struct VdcCollection {
+    storage: Box<dyn StorageManagerInterface>,
+}
+
+pub enum VdcCollectionError {
+    SerializeFailed,
+    DeserializeFailed,
+    StoreFailed,
+    LoadFailed,
```

Comment:
You can capture the previous error for better context:
```suggestion
    StoreFailed(StorageManagerError),
    LoadFailed(StorageManagerError),
```

Reply:
Done.